### ISSUE TITLE
`to_plotly_json` does not handle `np.array` properly

### ISSUE THREAD

**Issue by ChiQiao** [plotly/plotly.py#4281]
Code to reproduce the issue:
```python
import json
import plotly.graph_objects as go
import numpy as np

fig = go.Figure(go.Scatter(x=np.array([0, 1]), y=np.array([0, 1])))
json.dumps(fig.to_plotly_json())  # TypeError: Object of type ndarray is not JSON serializable
```

**Comment by itsluketwist** [comment]
@ChiQiao - Hey! Whilst the `to_plotly_json` method doesn't provide a `dict` that is immediately `json` compatible, if you use the `PlotlyJSONEncoder` as the `cls` arg when creating the `json` then everything should work fine. 😄 

Fixed code snippet:

```
import json
import plotly.graph_objects as go
import numpy as np

from plotly.utils import PlotlyJSONEncoder  # need this import

fig = go.Figure(go.Scatter(x=np.array([0, 1]), y=np.array([0, 1])))
json.dumps(fig.to_plotly_json(), cls=PlotlyJSONEncoder)  # encoder used as arg here, handles the numpy data  (amongst others)
```

Hopefully this fixes your issue!

**Comment by ChiQiao** [comment]
@itsluketwist Thanks for the information. Do you think this is the expected behavior (i.e., fulfilling the purpose of this method as `Convert figure to a JSON representation as a Python dict`)? If so, please feel free to close this issue.

**Comment by itsluketwist** [comment]
@ChiQiao - agreed, this experience could be better. I've made a PR to hopefully make it so, and discuss the json usage with project maintainers. 😄

**Comment by itsluketwist** [comment]
@ChiQiao - I don't know the reason for the multiple very similar functions here, especially when there is also a `to_dict` method that is the same as `to_plotly_json`. 😅 

You're right that the `to_json` method should also fix your problem! 😄

**Comment by alexcjohnson** [comment]
Closed by #4301 - thanks @itsluketwist 🎉
